sensor/common/processsignal/pipeline.go

```diff
@@ -52,6 +75,43 @@ func populateIndicatorFromCachedContainer(indicator *storage.ProcessIndicator, c
 	indicator.ImageId = cachedContainer.ImageID
 }
 
+// Shutdown closes all communication channels and shutdowns the enricher
+func (p *Pipeline) Shutdown() {
+	p.cancelCurrentContext()
+	defer close(p.enrichedIndicators)
+}
+
+// Notify allows the component state to be propagated to the pipeline
+func (p *Pipeline) Notify(e common.SensorComponentEvent) {
+	log.Info(common.LogSensorComponentEvent(e))
+	switch e {
+	case common.SensorComponentEventCentralReachable:
+		p.createNewContext()
+	case common.SensorComponentEventOfflineMode:
+		p.cancelCurrentContext()
```

Comment:
It feels more natural for me to cancel on transition to offline, but I am open for arguments. 
Why did we cancel on reconnect in other places?

Reply:
At least that's how I did in the event pipeline, where we originally needed this behavior: https://github.com/stackrox/stackrox/blob/master/sensor/kubernetes/eventpipeline/pipeline_impl.go#L134

Reply:
Yeah. I think the only reason why it was done on reconnection with central in other components is because I had only one function to cancel the old context and create the new context, and in that case I needed to call it on reconnection.  https://github.com/stackrox/stackrox/blob/dbe7b6d5bac42ecc26b9f31ab4a5a4a1e4c7e350/sensor/common/networkflow/manager/manager_impl.go#L284-L291

I also think it's more clear this way; I'll open a PR to change in other components.